pandas/tests/frame/methods/test_diff.py

```diff
@@ -91,9 +91,13 @@ def test_diff_datetime_with_nat_zero_periods(self, tz):
 
         df[1] = ser.copy()
 
-        msg = "will attempt to set the values inplace instead"
-        with tm.assert_produces_warning(FutureWarning, match=msg):
-            df.iloc[:, 0] = pd.NaT
+        if tz is None:
+            msg = "will attempt to set the values inplace instead"
+            with tm.assert_produces_warning(FutureWarning, match=msg):
+                df.iloc[:, 0] = pd.NaT
```

Comment:
@phofl i think i misunderstood earlier when i said this _should_ warn.  i was under the impression that the RHS was `pd.Series([pd.NaT]*N, dtype="M8[ns]")`, not the scalar pd.NaT.  With the scalar, this should _not_ warn.

Reply:
that is an existing problem that i think is orthogonal to this PR, so no need to hold up on this.

Reply:
Yep agreed, the NaT gets cast to an all NaT Series without a timezone, hence this seems equal

Reply:
ideal then would be to assert_warns(None) unconditionally and xfail the tzaware case

Reply:
Done